cmake/GzUtils.cmake

```diff
@@ -1971,3 +1978,170 @@ macro(_gz_cmake_parse_arguments prefix options oneValueArgs multiValueArgs)
   endif()
 
 endmacro()
+
+#################################################
+# gz_add_get_install_prefix_impl(GET_INSTALL_PREFIX_FUNCTION <get_install_prefix_function>
+#                                GET_INSTALL_PREFIX_HEADER <get_install_prefix_function>
+#                                OVERRIDE_INSTALL_PREFIX_ENV_VARIABLE <override_install_prefix_env_variable>)
+#
+# This macro adds to ${PROJECT_LIBRARY_TARGET_NAME} the implementation of
+# of the function passed by GET_INSTALL_PREFIX_FUNCTION and declared in
+# GET_INSTALL_PREFIX_HEADER .
+#
+# The defined functions implements a GET_INSTALL_PREFIX_FUNCTION that returns
+# the installation directory of the package (CMAKE_INSTALL_PREFIX at build time)
+# as following:
+# * if the library is shared and GZ_ENABLE_RELOCATABLE_INSTALL is ON, by extracting the
+#   location of the shared library via dladdr, and computing the corresponding
+#   install prefix from it
+# * if the library is static or GZ_ENABLE_RELOCATABLE_INSTALL is OFF, by using the exact
+#   value of CMAKE_INSTALL_PREFIX that was hardcoded in the library at compilation time
+#
+# As in some cases it is important to have the ability to control and change the value returned by
+# the GET_INSTALL_PREFIX_FUNCTION at runtime, in both cases the library returns the value of
```

Comment:
Could we add please the documentation for the `GET_INSTALL_PREFIX_FUNCTION`, `GET_INSTALL_PREFIX_HEADER` and `OVERRIDE_INSTALL_PREFIX_ENV_VARIABLE` using the same format that the rest of the GzUtils file? 

snippet from gz_build_executables
```cmake
#################################################
# gz_build_executables(SOURCES <sources>
#                       [PREFIX <prefix>]
#                       [LIB_DEPS <library_dependencies>]
#                       [INCLUDE_DIRS <include_dependencies>]
#                       [EXEC_LIST <output_var>]
#                       [EXCLUDE_PROJECT_LIB])
#
# Build executables for an Gazebo project. Arguments are as follows:
#
# SOURCES: Required. The names (without a path) of the source files for your
#          executables.
#
# PREFIX: Optional. This will append <prefix> onto each executable name.
#
```
```